blazar/plugins/devices/device_plugin.py

```diff
@@ -100,6 +100,9 @@ def _get_plugins(self):
 
     def reserve_resource(self, reservation_id, values):
         """Create reservation."""
+        reservation = db_api.reservation_get(reservation_id)
+        lease = db_api.lease_get(reservation["lease_id"])
+        values["project_id"] = lease["project_id"]
```

Comment:
By this point, the system will have already fetched the lease, so it is a bit inefficient to fetch this data again.

https://github.com/ChameleonCloud/blazar/pull/42/files#diff-e058dc95ba9c43e758e0450030204290dee6909a02a4082ebf5d20590f1451b7R455-R459

What if the values dict gets the lease project_id added at that point ^ ?